redisinsight/api/src/modules/recommendation/providers/recommendation.provider.ts

```diff
@@ -411,6 +416,122 @@ export class RecommendationProvider {
     }
   }
 
+  /**
+  * Check search indexes recommendation
+   * @param redisClient
+   * @param keys
+   * @param client
+   */
+  // eslint-disable-next-line
+  async determineSearchIndexesRecommendation(
+    redisClient: Redis,
+    keys: Key[],
+    client: any,
+  ): Promise<Recommendation> {
+    try {
+      if (client.isCluster) {
+        let processedKeysNumber = 0;
+        let isJSONOrHash = false;
+        let sortedSetNumber = 0;
+        while (
+          processedKeysNumber < keys.length
+          && !isJSONOrHash
+          && sortedSetNumber <= sortedSetCountForCheck
+        ) {
+          if (keys[processedKeysNumber].type !== RedisDataType.ZSet) {
+            processedKeysNumber += 1;
+          } else {
+            let keyType: string;
+            const sortedSetMember = await redisClient.sendCommand(
+              new Command('zrange', [keys[processedKeysNumber].name, 0, 0], { replyEncoding: 'utf8' }),
+            ) as string[];
+            try {
+              keyType = await redisClient.sendCommand(
+                new Command('type', [sortedSetMember[0]], { replyEncoding: 'utf8' }),
+              ) as string;
+            } catch (err) {
+              if (err && checkRedirectionError(err)) {
+                const { address } = parseRedirectionError(err);
+                const nodes = client.nodes('master');
+
+                const node: any = nodes.find(({ options: { host, port } }: Redis) => `${host}:${port}` === address);
+                if (!node) {
+                  throw new ClusterNodeNotFoundError(
+                    ERROR_MESSAGES.CLUSTER_NODE_NOT_FOUND(node),
+                  );
+                }
+
+                keyType = await node.sendCommand(
+                  new Command('type', [sortedSetMember[0]], { replyEncoding: 'utf8' }),
+                ) as string;
+              }
+            }
+            if (keyType === RedisDataType.JSON || keyType === RedisDataType.Hash) {
+              isJSONOrHash = true;
+            }
+            processedKeysNumber += 1;
+            sortedSetNumber += 1;
+          }
+        }
```

Comment:
In general I would like to see 2 separate methods for cluster and standalone but with almost the same logic except pipeline vs Promise.all sendCommand (on cluster client). Is it possible?